client/src/main/java/io/pravega/client/stream/impl/Orderer.java

```diff
@@ -55,6 +55,7 @@ public class Orderer {
     @VisibleForTesting
     public <T extends EventSegmentReader> T nextSegment(List<T> segments) {
         if (segments.isEmpty()) {
+            log.debug("List of segments is empty.");
```

Comment:
Rather than logging this here, perhaps it would better to log in the caller where more context is available.

Reply:
Done.